test/suite/Annotations.test.php

```diff
@@ -585,12 +590,149 @@ protected function testGetAnnotationsFromTrait()
+        $this->check($annotations[1]->note === 'inheritance-trait-tester', 'child annotation second');
+    }
+
+    protected function testHandlesMethodAliasingWithTraits()
+    {
+        if (version_compare(PHP_VERSION, '5.4.0', '<')) {
+            $this->pass();
+            return;
+        }
+
+        $annotations = Annotations::ofMethod('AliasTraitTester', 'baseTraitRun');
+        $this->check(count($annotations) === 2, 'baseTraitRun inherits annotation');
+        $this->check($annotations[0]->note === 'alias-base-trait-tester', 'inherited annotation goes first');
+        $this->check($annotations[1]->note === 'alias-base-trait', 'non-inherited annotation goes second');
+
+        $annotations = Annotations::ofMethod('AliasTraitTester', 'traitRun');
+        $this->check(count($annotations) === 2, 'traitRun inherits annotation');
+        $this->check($annotations[0]->note === 'alias-base-trait-tester', 'inherited annotation goes first');
+        $this->check($annotations[1]->note === 'alias-trait', 'non-inherited annotation goes second');
+
+        $annotations = Annotations::ofMethod('AliasTraitTester', 'run');
+        $this->check(count($annotations) === 2, 'run inherits annotation');
+        $this->check($annotations[0]->note === 'alias-base-trait-tester', 'inherited annotation goes first');
+        $this->check($annotations[1]->note === 'alias-trait-tester', 'non-inherited annotation goes second');
+    }
+
+    protected function testHandlesConflictedMethodSelectionWithTraits()
+    {
+        if (version_compare(PHP_VERSION, '5.4.0', '<')) {
+            $this->pass();
+            return;
+        }
+
+        $annotations = Annotations::ofMethod('InsteadofTraitTester', 'baseTrait');
+        $this->check(count($annotations) === 2, 'baseTrait inherits annotation');
+        $this->check($annotations[0]->note === 'insteadof-base-trait-tester', 'inherited annotation goes first');
+        $this->check($annotations[1]->note === 'insteadof-base-trait-b', 'non-inherited annotation goes second');
+
+        $annotations = Annotations::ofMethod('InsteadofTraitTester', 'trate');
+        $this->check(count($annotations) === 2, 'trate inherits annotation');
+        $this->check($annotations[0]->note === 'insteadof-base-trait-tester', 'inherited annotation goes first');
+        $this->check($annotations[1]->note === 'insteadof-trait-a', 'non-inherited annotation goes second');
+    }
+
+    protected function testCanGetPropertyAnnotationsIncludedFromTrait()
+    {
+        if (version_compare(PHP_VERSION, '5.4.0', '<')) {
+            $this->pass();
+            return;
+        }
+
+        $annotations = Annotations::ofProperty(new \ReflectionClass('SimpleTraitTester'), 'sampleFromTrait');
+        $this->check(count($annotations) > 0, 'from class reflection and property name');
+
+        $annotations = Annotations::ofProperty(new \ReflectionProperty('SimpleTraitTester', 'sampleFromTrait'));
```

Comment:
This and other checks below in this test method are redundant, because here you're testing how `Annotations::ofProperty` method works instead of what happens with traits.